pkg/solana/chainwriter/chain_writer.go

```diff
@@ -316,10 +402,47 @@ func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contra
 	discriminator := GetDiscriminator(methodConfig.ChainSpecificName)
 	encodedPayload = append(discriminator[:], encodedPayload...)
 
+	// Fetch latest blockhash
+	blockhash, err := s.client.LatestBlockhash(ctx)
+	if err != nil {
+		return errorWithDebugID(fmt.Errorf("error fetching latest blockhash: %w", err), debugID)
+	}
+
+	if len(createATAinstructions) > 0 {
```

Comment:
nit: Optional suggestion.
This function is becoming too big to read. Maybe same for chain_writer file?
Maybe move this if block out of this function, to another function handlePossibleATACreation()
Maybe even create a new go file ata_creation.go

Reply:
Broke this out into a separate function in a different file